CMakeLists.txt

```diff
@@ -18,6 +18,8 @@ endif()
 message(STATUS "Build configuration: " ${CMAKE_BUILD_TYPE})
 
 option(ASTE_SET_MESH_BLOCK "Use block versions to set mesh primitives." ON)
+option(ASTE_NN_GRADIENT_MAPPING "Support nearest neighbour gradient mapping." ON)
```

Comment:
In order to preserve backward compatibility, we could enable the support if version 2.4 was found by CMake, similar to these lines https://github.com/davidscn/matrix-free-dealii-precice/blob/bd991b980d29b01092b448f8d156d8c6f521c076/CMakeLists.txt#L108-L114